[CB-53] Allocated Objects Strategy

EPecherkin opened the pull request:
[CB-53](https://youtrack.toptal.net/youtrack/issue/CB-53)

jaimerson commented on features/change_files_spec.rb:
```diff
@@ -16,29 +16,48 @@ class Class1
+      end
+    RUBY
+
+    is_expected.to include(
```
Why is it `include` now instead of `match_array`? Shouldn't it remain strict?

EPecherkin replied:
I don't think so. It is not that bad if we predict a bit more. So, I think it is better to specify examples which should contain than to specify all output of predictor

jaimerson replied:
Maybe, but these specs allow it to predict a lot more. Is the full output inconveniently longer?

jaimerson reviewed (approved):
lgtm

pluff commented:
Please squash everything into meaningful commits

Commit:
Use helpers to fiter and transform paths in detectors

Commit:
Make file_paths uniq for case_map

Commit:
Improve features to cover more problems

Commit:
Strategy to track allocated objects during example

It uses ObjectSpace to fetch new objects allocated during
example and TracePoint to find places of its definition
